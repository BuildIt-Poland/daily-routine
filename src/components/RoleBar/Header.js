import styled from 'styled-components';

import { spacingMedium } from '../../styles/designTokens';

const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 1.4rem 0 ${spacingMedium};
  height: 5rem;
  background-color: rgba(255, 255, 255, 0.5);
`;

export default Header;
