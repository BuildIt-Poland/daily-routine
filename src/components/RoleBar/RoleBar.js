import React from 'react';
import PropTypes from 'prop-types';

import Subheadline from '../Subheadline';
import Header from './Header';
import SlidingPanel from '../SlidingPanel';

function RoleBar({ label }) {
  return (
    <Header>
      <Subheadline>
        I am <strong>{label}</strong>
      </Subheadline>
      <SlidingPanel />
    </Header>
  );
}

RoleBar.propTypes = {
  label: PropTypes.string.isRequired
};

export default RoleBar;
