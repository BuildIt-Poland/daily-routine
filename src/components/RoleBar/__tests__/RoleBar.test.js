import React from 'react';
import { create } from 'react-test-renderer';

import RoleBar from '../RoleBar';

describe('COMPONENT - RoleBar', () => {
  it('renders correctly', () => {
    const component = create(<RoleBar />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
