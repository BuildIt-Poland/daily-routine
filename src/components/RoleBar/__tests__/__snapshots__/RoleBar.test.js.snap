// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - RoleBar renders correctly 1`] = `
.c1 {
  font-size: 1.8rem;
  margin: 1rem 0;
  font-weight: 400;
}

.c2 {
  border: 0;
  margin: 0;
  padding: 0;
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 4rem;
  min-height: 4rem;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 1.4rem 0 2rem;
  height: 5rem;
  background-color: rgba(255,255,255,0.5);
}

<header
  className="c0"
>
  <h2
    className="c1"
  >
    I am 
    <strong>
      Frond End Developer
    </strong>
  </h2>
  <button
    className="c2"
    onClick={[Function]}
  >
    <GearIcon />
  </button>
</header>
`;
